software/authbox/badgereader_wiegand_gpio.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 Ace Monster Toys. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wiegand based badge reader directly connected via GPIO
+"""
+
+from authbox.api import BaseWiegandPinThread, GPIO
+import time
+
+
+class WiegandGPIOReader(BaseWiegandPinThread):
+  """Badge reader hardware abstraction.
+
+  A Wiegand GPIO badge reader is defined in config as:
+
+    [pins]
+    name = WiegandGPIOReader:3:5
+
+  where 3 is the D0 pin (physical numbering), and 5 is the D1 pin (also 
+  physical numbering).
+  """
+
+  # This is a class variable to store the bitstream transmitted by the reader.
+  # It is stored using a string to avoid byte ordering issues as well as to
+  # simplify its processing.  It was initially implemented using the
+  # "bitstream" package, but this then had a dependency on a C compiler and
+  # NumPy
+  bits = ''
```

Comment:
I would rather see bits and timeout treated as instance variables (set in `__init__`) and DEFAULT_TIMEOUT_IN_MS as a module constant.  In your experimenting, how many ms apart were the bits (that's worth documenting in a comment)...